First commit UI tests

Smetankina opened the pull request:
Ui

Commit:
First commit UI tests

alesyaisachenko commented on pom.xml:
```diff
@@ -0,0 +1,49 @@
+
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
```
junit и testng -- это тестовые фреймворки, в проекте используют только один тестовый фреймворк (или JUnit или TestNG).

удали, плиз, настройку на TestNG

Smetankina replied:
удалила

alesyaisachenko commented on src/main/java/ru/yandex/practikum/HomePageClass.java:
```diff
@@ -0,0 +1,187 @@
+    //кнопка Заказать в ThirdPart
+    private By orderButtonInEnd = By.className("Button_Button__ra12g Button_Middle__1CSJM");
+    //0 Кликнуть на "Сколько это стоит?И как оплатить?" By.id="accordion__heading-0"
+    private By List0 = By.id("accordion__heading-0");
```
List0 -- нужно list0 , названия для переменных нужно задавать в lowerCamelCase.

в Java есть строгие правила к именованию пакетов, классов, методов, переменных. 

Шпаргалка по неймингу: - классы - имя начинается с заглавной буквы, если в имени несколько слов, каждое слово пишется с заглавной буквы слитно (пример: Praktikum, BestProgram) - методы - именуются lowerCamelCase - переменные - именуются lowerCamelCase, нельзя использовать цифры в качестве первого символа (пример: firstName, expectedResult, tylerTheCreator. ) - пакеты - допустимы только строчные буквы - константы - Константы состоят из заглавных символов, если в имени несколько слов то они разделяются нижним подчёркиванием "_".

alesyaisachenko commented on src/main/java/ru/yandex/practikum/HomePageClass.java:
```diff
@@ -0,0 +1,187 @@
+    //0 Кликнуть на "Сколько это стоит?И как оплатить?" By.id="accordion__heading-0"
+    private By List0 = By.id("accordion__heading-0");
+    //Сутки — 400 рублей. Оплата курьеру — наличными или картой.
+    private By textList0 = By.xpath(".//*[@id='accordion__panel-0']/p");
```
textList0 -- а теперь замечание по поводу использования чисел. При нейминге не рекомендуется использовать числа, их еще называют magicNumbers. Очень тяжело поддерживать и читается код с magicNumbers.

Лучше привязываться к теме вопроса \ ответа.

Подбери, пожалуйста, более удачные названия для всех переменных формата List и textList.

Smetankina replied:
Я  тогда в классе QuestionImportantTest тоже переименую переменные textExpected0 и textActual0, верно же?

alesyaisachenko replied:
да, все верно, принцип тот же

alesyaisachenko replied:
вообще есть хорошая книга («Чистый код» Р. Мартин), в ней описаны правила написания красивого кода. Первая глава как раз посвящена правильному неймингу.

Советую ее почитать

alesyaisachenko commented on src/main/java/ru/yandex/practikum/OrderPageClass.java:
```diff
@@ -0,0 +1,146 @@
+    //Срок аренды
+    public OrderPageClass clickRentPeriod() {
+        driver.findElement(rentPeriodField).click();
+        driver.findElement(By.xpath(".//*[(@role ='option' and text()='сутки')]")).click();
```
By.xpath(".//*[(@role ='option' and text()='сутки')]") -- выдели, пожалуйста, в отдельный элемент.

В методах мы работаем с переменными, не должно быть XPath

Smetankina replied:
Вынесла в переменную rentForADay

alesyaisachenko commented on src/main/java/ru/yandex/practikum/OrderPageClass.java:
```diff
@@ -0,0 +1,146 @@
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeDriver;
```
не используемые импорты нужно удалять

Smetankina replied:
готово

alesyaisachenko commented on src/test/java/scootertests/Chrome/TestBase.java:
```diff
@@ -0,0 +1,39 @@
+      //  driver = new ChromeDriver(options);
+      driver=new ChromeDriver();
+        driver.manage().timeouts().implicitlyWait(7,TimeUnit.SECONDS);
+
```
не нужно было дублировать тесты для разных браузеров.

можно было просто в этом месте написать код для запуска Firefox и закомментировать его.

Smetankina replied:
сделано

alesyaisachenko commented on src/test/java/scootertests/Chrome/QuestionImportantTest.java:
```diff
@@ -0,0 +1,134 @@
+
+    @Test
+
+    public void test0() {
```
test0() -- не удачный нейминг с числами. Из названия теста должно быть ясно чем один тест отличается от другого.

Smetankina replied:
исправила

alesyaisachenko commented on src/main/java/ru/yandex/practikum/HomePageClass.java:
```diff
@@ -0,0 +1,187 @@
+    // Да, обязательно. Всем самокатов! И Москве, и Московской области.
+    private By textList7 = By.xpath(".//*[@id='accordion__panel-7']/p");
+
+
```
Советую использовать сочетание клавиш Ctrl+Alt+L для расстановки пробелов и табуляции в соответствии с настройками Code Style в Idea.

Smetankina replied:
прощлась по всем классам, спасибо

Commit:
Second commit UI tests

1. Удалила зависимость на TestNG
2. Переименованы переменные и методы по смыслу строчек, с маленькой буквы CamelCase
3.Вынесла в переменную rentForADay OrderPageClass
4. Убрала неиспользуемые импорты
5. Добавила в TestBase код для запуска Firefox и закомментировала его; дубли классов удалила.
6. Прошлась по классам, отформатировала код через hot keys
